apps/agent/app/src/components/Transactions.js

```diff
@@ -0,0 +1,476 @@
+  IconExternal,
+  IconToken,
+  IconLabel,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { useConnectedAccount, useNetwork } from '@aragon/api-react'
+import AgentSvg from './assets/agent_badge.svg'
+import EmptyTransactions from './EmptyTransactions'
+import { useIdentity } from './IdentityManager/IdentityManager'
+import LocalIdentityBadge from './LocalIdentityBadge/LocalIdentityBadge'
+import TransactionFilters from './TransactionFilters'
+import {
+  TRANSACTION_TYPES_LABELS,
+  TRANSACTION_TYPES_STRING,
+} from '../transaction-types'
+import useFilteredTransactions from './useFilteredTransactions'
+import useDownloadData from './useDownloadData'
+import { formatTokenAmount } from '../lib/utils'
+import { addressesEqual, toChecksumAddress } from '../lib/web3-utils'
+
+const getTokenDetails = (details, { address, decimals, symbol }) => {
+  details[toChecksumAddress(address)] = {
+    decimals,
+    symbol,
+  }
+  return details
+}
+
+const Transactions = React.memo(function Transactions({
+  agentAddress,
+  tokens,
+  transactions,
+}) {
+  const connectedAccount = useConnectedAccount()
+  const { layoutName } = useLayout()
+  const network = useNetwork()
+  const theme = useTheme()
+  const compactMode = layoutName === 'small'
+  const {
+    emptyResultsViaFilters,
+    filteredTransactions,
+    handleClearFilters,
+    handleSelectedDateRangeChange,
+    handleTokenChange,
+    handleTransactionTypeChange,
+    page,
+    setPage,
+    selectedDateRange,
+    selectedToken,
+    selectedTransactionType,
+  } = useFilteredTransactions({ transactions, tokens })
+  const tokenDetails = tokens.reduce(getTokenDetails, {})
+  const { onDownload } = useDownloadData({
+    filteredTransactions,
+    tokenDetails,
+    tokens,
+    selectedDateRange,
+  })
+  const symbols = tokens.map(({ symbol }) => symbol)
+
+  const sortedTransactions = useMemo(
+    () =>
+      filteredTransactions.sort(({ date: dateLeft }, { date: dateRight }) =>
+        // Sort by date descending
+        compareDesc(dateLeft, dateRight)
+      ),
+    [filteredTransactions]
+  )
+
+  if (!transactions.length) {
+    return <EmptyTransactions />
+  }
+
+  return (
+    <DataView
+      entries={sortedTransactions}
+      fields={
+        emptyResultsViaFilters
+          ? []
+          : [
+              { label: 'Date', priority: 2 },
+              { label: 'Source/recipient', priority: 5, childStart: true },
+              { label: 'Type', priority: 3 },
+              { label: 'Reference', priority: 1 },
+              { label: 'Amount', priority: 4 },
+            ]
+      }
+      heading={
+        <React.Fragment>
+          <div
+            css={`
+              padding: ${2 * GU}px 0;
+              display: flex;
+              align-items: center;
+              justify-content: space-between;
+            `}
+          >
+            {!compactMode && (
+              <TransactionFilters
+                compactMode={compactMode}
```

Comment:
Hmmm, I don't quite get it; this boolean will be `true` if `layoutName === 'small'`.

Reply:
Sorry I meant the opposite: this will always be `false` :smile: 

This is because of the `{!compactMode && (` that unmounts the component when `compactMode` is `true`.

Reply:
Aaah, gotcha!